build fails Ubuntu 20.23 - fmt/core.h:1706:15: error: possibly dangling reference to a temporary

```
[ 38%] Linking CXX static library ../../lib/libkp_logger.a
[ 38%] Built target kp_logger
[ 44%] Building CXX object src/CMakeFiles/kompute.dir/Algorithm.cpp.o
[ 50%] Building CXX object src/CMakeFiles/kompute.dir/Manager.cpp.o
In file included from /home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/format.h:48,
                 from /home/rap/arx86/kompute/src/include/kompute/Algorithm.hpp:6,
                 from /home/rap/arx86/kompute/src/include/kompute/operations/OpAlgoDispatch.hpp:4,
                 from /home/rap/arx86/kompute/src/include/kompute/Sequence.hpp:6,
                 from /home/rap/arx86/kompute/src/include/kompute/Manager.hpp:9,
                 from /home/rap/arx86/kompute/src/Manager.cpp:3:
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h: In instantiation of ‘constexpr fmt::v8::detail::value<Context> fmt::v8::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; type <anonymous> = fmt::v8::detail::type::custom_type; T = fmt::v8::join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*> >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*> >, char>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’:
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h:1855:77:   required from ‘constexpr fmt::v8::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {fmt::v8::join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, char>&}; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {fmt::v8::join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, char>}]’
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h:1872:38:   required from ‘constexpr fmt::v8::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<Args>::type>::type ...> fmt::v8::make_format_args(Args&& ...) [with Context = basic_format_context<appender, char>; Args = {join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, char>&}]’
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h:3119:44:   required from ‘std::string fmt::v8::format(format_string<T ...>, T&& ...) [with T = {join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, char>}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*, std::allocator<const char*> > >, char> >]’
/home/rap/arx86/kompute/src/Manager.cpp:425:9:   required from here
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h:1706:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |               ^~~
/home/rap/arx86/kompute/build/_deps/fmt-src/include/fmt/core.h:1706:46: note: the temporary was destroyed at the end of the full expression ‘fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >().fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >::map<fmt::v8::join_view<__gnu_cxx::__normal_iterator<const char**, std::vector<const char*> >, __gnu_cxx::__normal_iterator<const char**, std::vector<const char*> >, char>&>((* & std::forward<fmt::v8::join_view<__gnu_cxx::__normal_iterator<const char**, vector<const char*> >, __gnu_cxx::__normal_iterator<const char**, vector<const char*> >, char>&>((* & val))))’
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/kompute.dir/build.make:90: src/CMakeFiles/kompute.dir/Manager.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:256: src/CMakeFiles/kompute.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

```

First of all, sorry for the late reply. I was a bit busy.
Second, here are some logs:

```
-- General purpose GPU compute framework built on Vulkan
-- =======================================================
--   KOMPUTE_OPT_BUILD_TESTS: OFF
--   KOMPUTE_OPT_ENABLE_BENCHMARK: OFF
--   KOMPUTE_OPT_CODE_COVERAGE: OFF
--   KOMPUTE_OPT_BUILD_DOCS: OFF
--   KOMPUTE_OPT_INSTALL: OFF
--   KOMPUTE_OPT_BUILD_PYTHON: OFF
--   KOMPUTE_OPT_LOG_LEVEL: Default
--   KOMPUTE_OPT_USE_SPDLOG: OFF
--   KOMPUTE_OPT_ANDROID_BUILD: OFF
--   KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS: OFF
--   KOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK: OFF
--   KOMPUTE_OPT_BUILD_SHADERS: OFF
--   KOMPUTE_OPT_USE_BUILT_IN_SPDLOG: ON
--   KOMPUTE_OPT_SPDLOG_ASYNC_MODE: OFF
--   KOMPUTE_OPT_USE_BUILT_IN_FMT: ON
--   KOMPUTE_OPT_USE_BUILT_IN_GOOGLE_TEST: ON
--   KOMPUTE_OPT_USE_BUILT_IN_PYBIND11: ON
--   KOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER: ON
--   KOMPUTE_OPT_BUILT_IN_VULKAN_HEADER_TAG: v1.3.231
-- =======================================================
-- Ensuring the currently installed driver supports the Vulkan version requested by the Vulkan Header.
-- Found Vulkan Header version: 1.3.231
-- Running vulkaninfo was successful. Parsing the output...
-- GPU0 supports Vulkan API version '1.3.289'.
Valid GPU (GPU0) for Vulkan header version 1.3.231 found. GPU0 supports up to Vulkan 1.3.289.
-- Module support is disabled.
-- Version: 8.1.1
-- Build type: 
-- CXX_STANDARD: 14
-- Required features: cxx_variadic_templates
-- Setting KOMPUTE_OPT_LOG_LEVEL to according to the build type
-- Configuring done (0.6s)
-- Generating done (0.0s)
-- Build files have been written to: /home/bhaia/code/kompute/examples/array_multiplication/build
~/code/kompute/examples/array_multiplication/build (master ✗) cmake --build .
[  8%] Built target shader
[ 16%] Built target kp_shader
[ 29%] Built target fmt
[ 33%] Building CXX object kompute_build/src/logger/CMakeFiles/kp_logger.dir/Logger.cpp.o
[ 37%] Linking CXX static library libkp_logger.a
[ 37%] Built target kp_logger
[ 41%] Building CXX object kompute_build/src/CMakeFiles/kompute.dir/Algorithm.cpp.o
In file included from /home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/format.h:48,
                 from /home/bhaia/code/kompute/src/include/kompute/Algorithm.hpp:6,
                 from /home/bhaia/code/kompute/src/Algorithm.cpp:4:
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h: In instantiation of ‘constexpr fmt::v8::detail::value<Context> fmt::v8::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; type <anonymous> = fmt::v8::detail::type::string_type; T = std::__cxx11::basic_string<char>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’:
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h:1855:77:   required from ‘constexpr fmt::v8::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {const char (&)[12], const char (&)[9], const char (&)[43], int&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {char [12], char [9], char [43], int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
 1853 |         data_{detail::make_arg<
      |               ~~~~~~~~~~~~~~~~~                                              
 1854 |             is_packed, Context,
      |             ~~~~~~~~~~~~~~~~~~~                                              
 1855 |             detail::mapped_type_constant<remove_cvref_t<T>, Context>::value>(
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 1856 |             std::forward<T>(args))...} {
      |             ~~~~~~~~~~~~~~~~~~~~~~                                           
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h:1872:38:   required from ‘void fmt::v8::print(format_string<T ...>, T&& ...) [with T = {const char (&)[12], const char (&)[9], const char (&)[43], int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}; format_string<T ...> = basic_format_string<char, const char (&)[12], const char (&)[9], const char (&)[43], int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >]’
 1872 |   return {std::forward<Args>(args)...};
      |                                      ^
/home/bhaia/code/kompute/src/Algorithm.cpp:36:9:   required from here
  145 |     fmt::print("[{} {}] [warn] [{}:{}] {}\n",                                  \
      |     ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  146 |                __DATE__,                                                       \
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  147 |                __TIME__,                                                       \
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  148 |                __FILE__,                                                       \
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  149 |                __LINE__,                                                       \
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  150 |                fmt::format(__VA_ARGS__))
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h:1706:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |               ^~~
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h:1706:46: note: the temporary was destroyed at the end of the full expression ‘fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >().fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >::map<std::__cxx11::basic_string<char> >((*(const std::__cxx11::basic_string<char>*)(& std::forward<__cxx11::basic_string<char>&>((* & val)))))’
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [kompute_build/src/CMakeFiles/kompute.dir/build.make:79: kompute_build/src/CMakeFiles/kompute.dir/Algorithm.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:394: kompute_build/src/CMakeFiles/kompute.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```


I'm more of a Python + JS guy, so I'm not 100% sure what's going on here. I think some dependencies might need to be updated for newer versions of GCC, but I'm having trouble with this C++ package management.

I'll get back to you if I figure it out.

Thanks so much!

This is what killed it for sure:

```
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h:1706:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |               ^~~
/home/bhaia/code/kompute/examples/array_multiplication/build/_deps/fmt-src/include/fmt/core.h
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,7 +189,7 @@ endif()
 if(KOMPUTE_OPT_USE_BUILT_IN_FMT)
     set(FMT_INSTALL ${KOMPUTE_OPT_INSTALL})
     FetchContent_Declare(fmt GIT_REPOSITORY https://github.com/fmtlib/fmt.git
-        GIT_TAG 8.1.1
+        GIT_TAG 11.0.0
         GIT_SHALLOW 1) # Source: https://github.com/fmtlib/fmt/releases
     FetchContent_MakeAvailable(fmt)
 else()

